src/components/form/form.js

```diff
@@ -348,6 +379,28 @@ class Form extends React.Component {
     this.activeInput = input;
   }
 
+  /**
+   * Sets the form to Dirty
+   *
+   * @method setIsDirty
+   * @return {void}
+   */
+  setIsDirty = () => {
+    this.isDirty = true;
+    this.setState({ isDirty: this.isDirty });
```

Comment:
could wrap this in a `if (!this.state.isDirty) {` to block unrequired calls to `setState`?